pkg/kubelet/cm/cpumanager/policy_static.go

```diff
@@ -156,9 +156,15 @@ func (p *staticPolicy) assignableCPUs(s state.State) cpuset.CPUSet {
 }
 
 func (p *staticPolicy) AddContainer(s state.State, pod *v1.Pod, container *v1.Container, containerID string) error {
-	glog.Infof("[cpumanager] static policy: AddContainer (pod: %s, container: %s, container id: %s)", pod.Name, container.Name, containerID)
 	if numCPUs := guaranteedCPUs(pod, container); numCPUs != 0 {
+		glog.Infof("[cpumanager] static policy: AddContainer (pod: %s, container: %s, container id: %s)", pod.Name, container.Name, containerID)
 		// container belongs in an exclusively allocated pool
+
+		if _, ok := s.GetCPUSet(containerID); ok {
```

Comment:
Do we still need this `if` block?

Reply:
Yes, this a semantic change in that Add must now be idempotent just like remove. @flyingcougar could you add a note about idempotency to the Policy interface for both Add and Remove?

Reply:
I understand. However, the runtime will not call `AddContainer` for containers already in state. And after the panic PR merges, the reconcile loop should not be calling `AddContainer` on a container already in state (see: https://github.com/kubernetes/kubernetes/pull/54410/files#diff-67c46e7cbb41ea5e675000a3298e6b32R249). But I'm good with making it idempotent.

Reply:
i think it will just prevent us from unnecessary calling 'AddContainer()' in `reconcile` for containers that are already in state.

Reply:
@ConnorDoyle  ok